packages/react-components/lib/doors/utils.ts

```diff
@@ -0,0 +1,75 @@
+import * as RmfModels from 'rmf-models';
+
+export enum DoorType {
+  SingleSwing = RmfModels.Door.DOOR_TYPE_SINGLE_SWING,
+  SingleSliding = RmfModels.Door.DOOR_TYPE_SINGLE_SLIDING,
+  SingleTelescope = RmfModels.Door.DOOR_TYPE_SINGLE_TELESCOPE,
+  DoubleSwing = RmfModels.Door.DOOR_TYPE_DOUBLE_SWING,
+  DoubleSliding = RmfModels.Door.DOOR_TYPE_DOUBLE_SLIDING,
+  DoubleTelescope = RmfModels.Door.DOOR_TYPE_DOUBLE_TELESCOPE,
+}
+
+export enum DoorMotion {
+  Clockwise = 1,
+  AntiClockwise = -1,
+}
+
+export enum DoorMode {
+  Open = RmfModels.DoorMode.MODE_OPEN,
+  Closed = RmfModels.DoorMode.MODE_CLOSED,
+  Moving = RmfModels.DoorMode.MODE_MOVING,
+}
+
+export interface DetailedDoor extends RmfModels.Door {
```

Comment:
I would prefer to avoid making these extension to the models. It creates confusion where some places uses  `DetailedDoor` and others using the base `Door`. Sometimes its also confusing how can I create a `DetailedDoor`.

Some workaround:
* Instead of inheritance, use composition, and name it something like `DoorData`, with type `{ door: Door, level: string }`. At least to me, it's more obvious that you can create the object by mapping the building map level with the doors. Inheritance can mean that `Door` needs to be modified or processed in some way while composition makes it more obvious that you can use the `Door` from building map "as is".
* For cases where it only needs one door, just have 2 props, the `Door` and the level.
* For cases where you need an array, one solution is to use a tuple `[level: string, door: Door]`, but it isn't as readable as using composition (actually this is also a composition in a way).